File: frontend/src/app/core/services/location.service.ts
/**
 * Service für den Zugriff auf Location-Daten.
 *
 * Ruft die verfügbaren Städte mit Events vom Backend ab.
 */
import { Injectable, inject } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { environment } from "../../../environments/environment";

@Injectable({ providedIn: "root" })
export class LocationService {
  /** HTTP-Client für API-Aufrufe. */
  private readonly http = inject(HttpClient);

  /**
   * Ruft alle Städte ab, in denen Events stattfinden.
   *
   * @returns Observable mit der alphabetisch sortierten Städte-Liste
   */
  getCities(): Observable<string[]> {
    return this.http.get<string[]>(`${environment.apiUrl}/locations/cities`);
  }
}

File: backend/src/main/java/ch/loete/backend/domain/service/LocationService.java
package ch.loete.backend.domain.service;

import ch.loete.backend.process.repository.LocationRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * Service für Location-bezogene Operationen.
 *
 * <p>Stellt Methoden zum Abrufen der verfügbaren Städte bereit, in denen Events stattfinden.
 */
@Service
@RequiredArgsConstructor
public class LocationService {

  /** Repository für den Zugriff auf Location-Daten. */
  private final LocationRepository locationRepository;

  /**
   * Gibt alle Städte zurück, in denen mindestens ein Event stattfindet.
   *
   * @return alphabetisch sortierte Liste der Staedtenamen
   */
  @Transactional(readOnly = true)
  public List<String> getCities() {
    return locationRepository.findDistinctCitiesWithEvents();
  }
}
